pybigtools/src/lib.rs

```diff
@@ -1129,27 +1196,38 @@ fn open_path_or_url(
-///     names (None, bool, or int):  
-///         If `None`, then each return value will be a single `float`,
-///             the average value over an interval in the bed file.  
-///         If `True`, then each return value will be a tuple of the value of column 4
-///             and the average value over the interval with that name in the bed file.  
+///     names (None, bool, or int):
+///         If `None`, then each return value will be a single `float` or `int`,
+///             the calculated statistics value over an interval in the bed file.
+///         If `True`, then each return value will be a tuple of the value of column 4 and the
+///             calculated statistics value over the interval with that name in the bed file.
 ///         If `False`, then each return value will be a tuple of the interval in the format
-///             `{chrom}:{start}-{end}` and the average value over that interval.  
-///         If `0`, then each return value will match as if `False` was passed.  
+///             `{chrom}:{start}-{end}` and the calculated statistics value over that interval.
+///         If `0`, then each return value will match as if `False` was passed.
 ///         If a `1+`, then each return value will be a tuple of the value of column of this
-///             parameter (1-based) and the average value over the interval.  
+///             parameter (1-based) and the calculated statistics value over the interval.
+///     stats (None, str): Calculate specific statistics for each bed entry:
+///         - `size`: Size of bed entry (`int`)
+///         - `bases`: Bases covered by bigWig (`int`)
+///         - `sum`: Sum of values over all bases covered (`float`)
+///         - `mean0`: Average over bases with non-covered bases counting as zeroes (`float`)
+///         - `mean` or `None`: Average over just covered bases (`float`)
+///         - `min`: Minimum over all bases covered (`float`)
+///         - `max`: Maximum over all bases covered (`float`)
+///         - `all`: `size`, `bases`, `sum`, `mean0`, `mean`, `min` and `max`
 ///
 /// Returns:
-///     This returns a generator of values. (Therefore, to save to a list, do `list(bigWigAverageOverBed(...))`)  
-///     If no name is specified (see the `names` parameter above), then returns a generator of `float`s.  
-///     If a name column is specified (see above), then returns a generator of tuples `({name}, {average})`  
+///     This returns a generator of values. (Therefore, to save to a list, do `list(bigWigAverageOverBed(...))`)
+///     If no name is specified (see the `names` parameter above), then returns a generator of `float`s or `int`s
+///     depending on the chosen statistics to calculate.
+///     If a name column is specified (see above), then returns a generator of tuples `({name}, {stats})`
 #[pyfunction]
 fn bigWigAverageOverBed(
     py: Python,
     bigwig: String,
     bed: String,
     names: Option<PyObject>,
+    stats: Option<String>,
```

Comment:
I wonder if it's worth allowing for a list of stats? Certainly, there will be an overhead to checking what stats to return vs just returning all the stats. But, could just leave that the API consumer.